### DISCUSSION TITLE
Support of rate limit?

### DISCUSSION THREAD

**Discussion by marcosflobo** [cloudbees/zendesk-java-client#729] [Q&A] [answered]
Hello team,

I would like to know if this SDK natively supports the pagination and the rate limit.

For example, if the rate limit is reached, should the user of the SDK handle the active wait?

Thank you!

**Comment by PierreBtz** [comment, marked as answer]
Hello

The pagination is handled, we are switching to CBP (cursor based pagination) the endpoints that support it. The rest of the endpoints are using offset pagination. See this documentation for details https://developer.zendesk.com/api-reference/introduction/pagination/.

Rate limit is handled as a specific exception that you can catch (see https://github.com/cloudbees-oss/zendesk-java-client/blob/465c7e8ba135f53821c8cf2c770a888a04d2a26c/src/main/java/org/zendesk/client/v2/ZendeskResponseRateLimitException.java), but you will have to handle the retries yourself.
